src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -486,7 +486,8 @@ public SqlDecimal(decimal value)
             // m_data1 = *pInt++; // lo part
             // m_data2 = *pInt++; // mid part
 
-            int[] bits = decimal.GetBits(value);
+            Span<int> bits = stackalloc int[4];
+            decimal.GetBits(value, bits);
```

Comment:
I don't think so.  It's hard-coded to return 4.  The tests for it validate it returns 4. It's documented to return 4.  If it doesn't return 4, it's a bug.  I'm not entirely clear on why we decided it must return int rather than void.  Regardless, I don't think we need to assert.  We don't, for example, assert in the original code that the array returned by GetBits had a length of 4.